src/main/kotlin/io/snyk/plugin/snykcode/core/PDU.kt

```diff
@@ -21,55 +24,56 @@
~import snyk.common.SnykError
~import java.util.function.Consumer
 
 class PDU private constructor() : PlatformDependentUtilsBase() {
 
-    override fun getProject(file: Any): Any = toPsiFile(file).project
+    override fun getProject(file: Any): Any {
+        require(file is SnykCodeFile) { "File $file must be of type SnykCodeFile but is ${file.javaClass.name}" }
+        return file.project
+    }
 
     override fun getProjectName(project: Any): String = toProject(project).name
 
-    override fun getFileName(file: Any): String = toPsiFile(file).virtualFile.name
+    override fun getFileName(file: Any): String = toVirtualFile(file).name
 
-    override fun getFilePath(file: Any): String = toPsiFile(file).virtualFile.path
+    override fun getFilePath(file: Any): String {
+        return toVirtualFile(file).path
+    }
 
-    override fun getDirPath(file: Any): String = toPsiFile(file).virtualFile.let {
+    override fun getDirPath(file: Any): String = toVirtualFile(file).let {
         it.parent?.path ?: it.path // root dir case
     }
 
     override fun getProjectBasedFilePath(file: Any): String {
-        val psiFile = toPsiFile(file)
-        // looks like we don't need ReadAction for this (?)
-        val absolutePath = psiFile.virtualFile.path
-        val projectPath = psiFile.project.basePath
-            ?: throw IllegalStateException("No Project base Path found for file $psiFile")
+        require(file is SnykCodeFile) { "File $file must be of type SnykCodeFile but is ${file.javaClass.name}" }
+        val virtualFile = file.virtualFile
+        val absolutePath = virtualFile.path
+        val projectPath = file.project.guessProjectDir()?.path
+            ?: throw IllegalStateException("No Project base Path found for file $file")
```

Comment:
Because Jetbrains recommends explicitly to use guessProjectDir() instead of baseDir in the JavaDoc

Reply:
```Tries to guess the "main project directory" of the project.
There is no strict definition of what is a project directory, since a project can contain multiple modules located in different places, and the .idea directory can be located elsewhere (making the popular Project.getBaseDir method not applicable to get the "project directory"). This method should be preferred, although it can't provide perfect accuracy either.```